core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -96,14 +96,16 @@
   /** A map of worker addresses to the most recent epoch time when client fails to read from it. */
   private Map<WorkerNetAddress, Long> mFailedWorkers = new HashMap<>();
 
-  protected FileInStream(URIStatus status, InStreamOptions options, FileSystemContext context) {
-    AlluxioConfiguration conf = context.getPathConf(new AlluxioURI(status.getPath()));
+  protected FileInStream(URIStatus status, InStreamOptions options, FileSystemContext context)
+      throws IOException {
+    mContext = context;
+    mContext.blockReinit();
```

Comment:
the concern here is that some other code paths might create FileInStreams, since the lock is reentrant, if FileInStream is created from openFile, this call should not block.